lib/processor.js

```diff
@@ -321,6 +321,9 @@ function adjustBlock(block) {
      * @returns {Message} The same message, but adjusted to the correct location.
      */
     return function adjustMessage(message) {
+        if (typeof message.line !== "number") {
+            return message;
```

Comment:
Nice thought about editor integrations, LGTM.

Should we just set `message.line = 1` then?

Reply:
Do you mean setting `message.line = 1` _instead of_ the early `return` and continuing with the rest of the function as if the message occurred on line 1? I hadn't thought of doing it that way, but it could work! What I had in mind when I left the original comment was keeping the early `return` as in `return { ...message, line: /* read from block.position */, column: /* here too */ }` so we could point the message to the opening code fence.

Reply:
I’d like to write like the following because it is much cleaner:

```js
if (!Number.isInteger(message.line)) {
  message = {
    ...message,
    line: 1,
    column: 0,
  };
}
```

But there is a lint error, so should we do fast return:

```js
if (!Number.isInteger(message.line)) {
  return {
    ...message,
   // not sure how to write the following, should `const lineInCode = message.line - leadingCommentLines;` be reused?
    line: blockStart  + 1,
    column: block.rangeMap[1].indent,
  };
}

```

Reply:
Good point about the error from overwriting `argument`. Early return it is. For position, what if we pointed to the opening code fence itself? That's probably easiest to implement since we don't have to do any math: `return { ...message, line: block.position.start.line, column: block.position.start.column };`

Reply:
@btmills Done, please review again.